components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/utils/InboundWebsocketProcessorUtil.java

```diff
@@ -0,0 +1,711 @@
+        info.setApplicationId(authenticationContext.getApplicationId());
+        info.setType(authenticationContext.getKeyType());
+        info.setApiPublisher(authenticationContext.getApiPublisher());
+        info.setApplicationName(authenticationContext.getApplicationName());
+        info.setConsumerKey(authenticationContext.getConsumerKey());
+        info.setEndUserName(authenticationContext.getUsername());
+        info.setApiTier(authenticationContext.getApiTier());
+        info.setGraphQLMaxDepth(authenticationContext.getGraphQLMaxDepth());
+        info.setGraphQLMaxComplexity(authenticationContext.getGraphQLMaxComplexity());
+
+        inboundMessageContext.setKeyType(info.getType());
+        inboundMessageContext.setInfoDTO(info);
+        inboundMessageContext.setAuthContext(authenticationContext);
+        inboundMessageContext.setInfoDTO(info);
+        return authenticationContext.isAuthenticated();
+    }
+
+    /**
+     * Authenticates JWT token in incoming GraphQL subscription requests.
+     *
+     * @param inboundMessageContext InboundMessageContext
+     * @return true if authenticated
+     * @throws APIManagementException if an internal error occurs
+     * @throws APISecurityException   if authentication fails
+     */
+    public static boolean authenticateGraphQLJWTToken(InboundMessageContext inboundMessageContext)
+            throws APIManagementException, APISecurityException {
+
+        AuthenticationContext authenticationContext;
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(inboundMessageContext.getTenantDomain(),
+                true);
+        JWTValidator jwtValidator = new JWTValidator(new APIKeyValidator(), inboundMessageContext.getTenantDomain());
+        authenticationContext = jwtValidator.
+                authenticateForGraphQLSubscription(inboundMessageContext.getSignedJWTInfo(),
+                        inboundMessageContext.getApiContext(), inboundMessageContext.getVersion());
+        return validateAuthenticationContext(authenticationContext, inboundMessageContext);
+    }
+
+    /**
+     * Authenticates JWT token in incoming Websocket handshake requests.
+     *
+     * @param inboundMessageContext InboundMessageContext
+     * @return true if authenticated
+     * @throws APIManagementException if an internal error occurs
+     * @throws APISecurityException   if authentication fails
+     */
+    public static boolean authenticateWSJWTToken(InboundMessageContext inboundMessageContext)
+            throws APIManagementException, APISecurityException {
+
+        AuthenticationContext authenticationContext;
+        JWTValidator jwtValidator = new JWTValidator(new APIKeyValidator(), inboundMessageContext.getTenantDomain());
+        authenticationContext = jwtValidator.
+                authenticateForWebSocket(inboundMessageContext.getSignedJWTInfo(),
```

Comment:
Can't we only pass inboundMessageContext to this function, so we can get the already set parameters inside the function itself?

Reply:
same answer as below

Reply:
+1